### PR TITLE
[ci] Run hyperdebug tests on the nightly CI

### PR DESCRIPTION
**Pull request by pamaury**
Add hyperdebug tests to the nightly CI. I don't know if they should added to the regular CI or nightlies, I am happy to change that depending on the feedback. I have done nightly run manually to show that it works.

### CODE REVIEW

File: ci/azure-pipelines-nightly.yml
Comment (jwnrt) [inline ci/azure-pipelines-nightly.yml:61]: I can see the other jobs using this, so I'm sure it correct, but how does `"xilinx/vivado/$(VIVADO_VERSION)"` work?

Doesn't `$(...)` run commands and `${...}` expand variables? I see `VIVADO_VERSION` is a `variable:` above, but does Azure convert that to a command?

File: ci/azure-pipelines-nightly.yml
Comment (jwnrt) [inline ci/azure-pipelines-nightly.yml:67]: Bazel Q: I see the bob job below is using this kind of query to select tests, but the `rom_e2e` job is using `test_tag_filters` - do these do different things?

Also, should we filter out tests tagged `broken`? (not that there are any right now)

**Review by jwnrt — COMMENTED** [review]
This looks sensible to me, thanks @pamaury. I've checked your manual nightly run and can see it's running 3 tests which all passed. Did you cancel the PR CI run, or did that fail?

Just added some questions about the Bazel since I'm not very familiar with how it works.

File: ci/azure-pipelines-nightly.yml
Comment (pamaury) [reply ci/azure-pipelines-nightly.yml:61]: This is an [azure syntax](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#understand-variable-syntax): `${{ }}` gets expanded by Azure at compile-time of the YAML, while `$()` gets expanded at runtime by Azure before the task runs. Admitedly it's a bit confusing because it's also a bash expression.

Comment (jwnrt) [reply ci/azure-pipelines-nightly.yml:61]: Ah okay, thank you

**Comment by jwnrt** [comment]
I see (only) one of the i2c tests failed in the latest manual nightly run - are these tests generally quite flaky?

**Comment by moidx** [comment]
Hi @jwnrt, thanks for putting this together! For SiVal we want to have the ability to start moving some of the `cw310` tests to use the `hyp310` configuration in CI as this is going to be the equivalent configuration for Silicon. I think it makes sense to enable the target in CI once we get the nightly enabled. This is related to [#19524](https://github.com/lowRISC/opentitan/pull/19524).

**Review by jwnrt — APPROVED** [review]

**Review by engdoreis — APPROVED** [review]
LGTM.

**Comment by jwnrt** [comment]
@engdoreis @pamaury are you wanting to sort out the flaky test before this merges?

**Comment by timothytrippel** [comment]
Yes, agreed with @moidx, we want the hyper310 tests to run in pre-submit CI too. Eventually, everything should be a hyper310 test probably, but initially we could start by defining another job (like the "CW310 Manufacturing Tests" job) that just runs the hyper310 tests in pre-submit.

Just following here, are there some test regressions that should be fixed as part of this PR? would be good to get hyper310 tests blocking in CI soon to unblock SiVal work.

**Comment by pamaury** [comment]
Alternatively, we can put them in the normal CI but marked as non-blocking

**Comment by timothytrippel** [comment]
> > initially we could start by defining another job (like the "CW310 Manufacturing Tests" job) that just runs the hyper310 tests in pre-submit.
> 
> Just to clarify @timothytrippel, by pre-submit do you mean the tests that run on each PR?
> 
yes pre-submit == all CI steps that happen before a PR is merged

> > Just following here, are there some test regressions that should be fixed as part of this PR? would be good to get hyper310 tests blocking in CI soon to unblock SiVal work.
> 
> I'm not aware of test regressions from running in Hyperdebug, but enabling them in nightly ASAP will help us find any. The ambient light test mentioned above only relies on the break-out-board of i2c and SPI devices and not Hyperdebug, I was mistaken.

Here is an example regression: https://github.com/lowRISC/opentitan/pull/19519#issuecomment-1716383575
By running `hyper310` tests in presubmit, we could avoid this.

**Comment by timothytrippel** [comment]
> Alternatively, we can put them in the normal CI but marked as non-blocking. Also the CI failures are due to ROM E2E tests and not hyperdebug test failures. So overall I think that if people approve, we could merge that and then look into adding this to th regular CI if we see no issue.

Yeah, I think that is a good first step, lets run them as non-blocking first, then submit follow up PRs to fix the test regressions and make it blocking asap. This will be important for SiVal.

Just to confirm @pamaury, we can merge this right now as is correct? Without causing a block-CI issue? i.e. this only changes to run on nightly?

**Comment by pamaury** [comment]
Yes this PR is safe to be merged

**Comment by timothytrippel** [comment]
Thanks @pamaury , next step is to get hyper310 tests running non-blocking in CI, then we can make them blocking.

**Comment by pamaury** [comment]
Yes I will create a PR for that

**Pull request merged**
